guide.md

```diff
@@ -0,0 +1,350 @@
+# A guide to rules_nixpkgs
+
+TODO: does use flake work with vanilla direnv? nix-direnv is just an improvement over that, i believe
+TODO: use the same nixpkgs.nix for nixpkgs_git_repository and shell.nix
+TODO: is the flake shell pure? because if this is run on a non-nixos system, and /usr/bin/gcc continues to be visible, that messes things up
+
+---
+
+Bazel is a modern build system valued for performant, correct, and hermetic
+builds. It has a few shortcomings, one of which is that that Bazel is only
+hermetic if certain conditions are met, and they aren't in many people's build
+setups. In practice, Bazel build products fail to be reproducible because Bazel
+silently depends on various things present in the global environment, like a
+C++ compiler. This ties into another issue, which is that external dependencies
+have to be Bazel-ified in order to be usable from within Bazel. For language
+ecosystems where there is widespread adoption of a package manager, this isn't
+much of an issue --- for instance, `pip` dependencies are taken care of by
+`rules_python`. However, in the case of C/C++ projects or with library headers,
+third-party dependencies can still be difficult to manage.
+
+The rules_nixpkgs ruleset enables the use of the extensive Nixpkgs package set
+from Bazel, allowing it to provision external dependencies via the Nix package
+manager. Nix provides strong hermeticity and reproducibility guarantees for all
+build products, which are defined in a language also called Nix. A sizeable
+collection of such package definitions is available in the Nixpkgs repository,
```

Comment:
Given the general introductory tone of this document, it feels like mentions of technologies and repos like Nixpkgs should have links?

Reply:
Added some links, but I'm not sure what the best place to link for Nix-the-package-manager/Nix-the-language/etc is... maybe nix.dev?